examples/jsm/controls/TransformControls.js

```diff
@@ -492,7 +484,31 @@ class TransformControls extends Object3D {
 
 				}
 
-				this.rotationAngle = this._offset.dot( _tempVector.cross( this.eye ).normalize() ) * ROTATION_SPEED;
+				_tempVector.cross( this.eye );
+
+				// When _tempVector is 0 after cross with this.eye the vectors are parallel and should use in-plane rotation logic.
+				if ( _tempVector.length() === 0 ) {
```

Comment:
Shouldn't this be compared with an epsilon?

Reply:
By epsilon, do you mean a small threshold value? That could certainly help when the camera and axis are not strictly parallel but are practically parallel. When this happens the resulting cross product is normalized making a seemingly random vector that is used as the reference for rotation. If we we were to use an epsilon (assuming I am understanding it correctly) we would probably need to update the line `this.rotationAxis.copy( this.eye );` in the in_plane rotation logic to be something like `_inPlaneRotation ? this.rotationAxis.copy( _unit[ axis ] ) : this.rotationAxis.copy( this.eye );`; to ensure you rotate around the intended axis rather than `this.eye` which may be very slightly off-axis.

Reply:
@LeviPesin Pinging you to get feedback on my previous response to your question.

Reply:
I'm not very sure, but I think using `this.eye` would be fine if it would be in any case within distance of epsilon from the axis?